.github/workflows/automated-release.yml

```diff
@@ -0,0 +1,63 @@
+name: Automated Release
+on:
+  workflow_dispatch:
+    inputs:
+      releaseBranch:
+        description: 'The existing branch name to release from, e.g. release-0.12'
+        required: true
+      releaseTag:
+        description: 'The release tag, e.g. v0.12.0-rc1'
+        required: true
+
+jobs:
+  prepare-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
+
+      - name: Checkout source code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ inputs.releaseBranch }}
+
+      - name: Install dependencies
+        run: |
+          go mod download
+
+      - name: Prepare Release
+        shell: bash
+        run: |
+          GOPATH=$(go env GOPATH)
+          KSERVE_PATH=$GOPATH/src/github.com/kserve/kserve
+          echo "KSERVE_PATH=$KSERVE_PATH" >> "$GITHUB_ENV"
+          mkdir -p $KSERVE_PATH
+          cp -a . $KSERVE_PATH
+          cd $KSERVE_PATH
+          export RELEASE_BRANCH=${{ inputs.releaseBranch }}
+          export RELEASE_TAG=${{ inputs.releaseTag }}
+          
+          # Bump Versions
+          make bump-version
+          ./hack/generate-install.sh $RELEASE_TAG
```

Comment:
Can we upload the generated yamls to the release artifacts ?

Reply:
I had to manually upload currently

Reply:
Yes, we can upload a list of files when making the release